### PR TITLE
adding the ability to change swagger url

### PR DESCRIPTION
**Pull request by doron2402**
- Adding the ability to customize the swagger end-point by passing `routePrefix` 
- use `/documentation` by default

@mcollina

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -26,6 +26,7 @@ fastify.register(require('fastify-swagger'), {
       description: 'testing the fastify swagger api',
       version: '0.1.0'
     },
+    routePrefix: '/api-docs',
```
Comment (mcollina) [inline README.md]: can you set this to the default?

File: dynamic.js [hunk 2]
```diff
@@ -27,7 +27,8 @@ module.exports = function (fastify, opts, next) {
   const security = opts.swagger.security || null
 
   if (opts.exposeRoute === true) {
-    fastify.register(require('./routes'))
+    const routePrefix = opts.routePrefix || '/documentation'
+    fastify.register(require('./routes'), { routePrefix })
```
Comment (mcollina) [inline dynamic.js]: you can pass `{ prefix }` instead. That's more aligned with the fastify way.

File: routes.js [hunk 3]
```diff
@@ -4,8 +4,9 @@ const fp = require('fastify-plugin')
 const path = require('path')
 
 function fastifySwagger (fastify, opts, next) {
+  const routePrefix = opts.routePrefix || '/documentation'
   fastify.route({
-    url: '/documentation/json',
+    url: `${routePrefix}/json`,
```
Comment (mcollina) [inline routes.js]: if you pass `{ prefix }`  as an option, then you just have to specify `/json` as your route. See https://github.com/fastify/fastify/blob/master/docs/Plugins.md#plugin-options. Same for every other route.

File: routes.js [hunk 4]
```diff
@@ -25,16 +26,16 @@ function fastifySwagger (fastify, opts, next) {
   })
 
   fastify.route({
-    url: '/documentation',
+    url: `${routePrefix}`,
     method: 'GET',
     schema: { hide: true },
-    handler: (request, reply) => reply.redirect('./documentation/')
+    handler: (request, reply) => reply.redirect(`.${routePrefix}/`)
```
Comment (mcollina) [inline routes.js]: I'm not 100% sure how `reply.redirect()` works with the prefix. Open an issue on fastify if it does not work.

File: README.md (hunk 1 above)
Comment (doron2402) [reply README.md]: 👍

File: dynamic.js [hunk 5]
```diff
@@ -27,7 +27,8 @@ module.exports = function (fastify, opts, next) {
   const security = opts.swagger.security || null
 
   if (opts.exposeRoute === true) {
-    fastify.register(require('./routes'))
+    const prefix = opts.routePrefix || '/api-docs'
+    fastify.register(require('./routes'), { prefix })
```
Comment (doron2402) [inline dynamic.js]: @mcollina 
I did not have time today to dive in why this one is not working. but for some reason when passing `{ prefix: '/api-docs' }` tests are failing and I had to manually use the `opts.prefix`

Comment (doron2402) [reply dynamic.js]: I was expecting that the `prefix` attribute will automatically add a prefix and I wouldn't need to add `opts.prefix` to the routes.

Comment (doron2402) [reply dynamic.js]: I think I got it, I was expecting that `prefix` will be used inside of a plugin (same as a route).

**Comment by doron2402** [comment]
Hi @mcollina 
Should we wait until we fix the bug with the `prefix` before merging this pr?
thanks.

File: package.json [hunk 6]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "fastify-swagger",
-  "version": "0.8.3",
+  "version": "0.8.4",
```
Comment (mcollina) [inline package.json]: the maintainer bumps the version number. Can you please revert this change?

File: routes.js [hunk 7]
```diff
@@ -5,7 +5,7 @@ const path = require('path')
 
 function fastifySwagger (fastify, opts, next) {
   fastify.route({
-    url: '/documentation/json',
+    url: `${opts.prefix}/json`,
```
Comment (mcollina) [inline routes.js]: this should not be needed.

File: static.js [hunk 8]
```diff
@@ -35,7 +35,8 @@ module.exports = function (fastify, opts, next) {
   fastify.decorate('swagger', swagger)
 
   if (opts.exposeRoute === true) {
-    fastify.register(require('./routes'))
+    const prefix = opts.routePrefix || '/api-docs'
```
Comment (mcollina) [inline static.js]: the default should be `/documentation`.

**Comment by mcollina** [comment]
I've left some notes on the PR.

> Should we wait until we fix the bug with the prefix before merging this pr?


Which one?

File: routes.js (hunk 7 above)
Comment (doron2402) [reply routes.js]: seems like route prefixing doesn't work when wrapping the plugin with `fastify-plugin`

File: routes.js [hunk 9]
```diff
@@ -5,7 +5,7 @@ const path = require('path')
 
 function fastifySwagger (fastify, opts, next) {
   fastify.route({
-    url: `${opts.prefix}/json`,
+    url: '/json',
```
Comment (doron2402) [inline routes.js]: @mcollina Seems like when using `fastify-plugin` it's ignoring the route prefix

Comment (mcollina) [reply routes.js]: there is no need to use fastify-plugin in ./routes.js

Comment (doron2402) [reply routes.js]: Got it, I was confused why it was used there to begin with.

I'll take it out.

**Comment by doron2402** [comment]
ok @mcollina i think it's finally ready :)
Thanks for the comments 👍
